nixos/modules/services/misc/gitolite.nix

```diff
@@ -55,10 +116,109 @@ in
           Gitolite user account. This is the username of the gitolite endpoint.
         '';
       };
+
+      mutable = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Enables configuration of gitolite through nix.
+          This disables the gitolite-admin repo and enables config, keys, rc and customFiles.
+        '';
+      };
+
+      customFiles = mkOption {
+        type = types.nullOr (types.listOf (types.submodule {
+          options = {
+            path = mkOption {
+              type = types.str;
+              description = ''
+                Location of the custom defined file.
+              '';
+            };
+            file = mkOption {
+              type = types.path;
+              description = ''
+                Custom defined file.
+              '';
+            };
+          };
+        }));
+        description = ''
+          custom files to lay into the configured dataDir. Will override any files which are specified here.
+          Use with care!
+        '';
+        example = ''
+          {
+            name = ".gitolite/conf/irc-announce.conf"
+            content =
+              #!/bin/sh
+              contents of some file
+              ...
+          }
+        '';
+        default = null;
+      };
+
+      keys = mkOption {
+        type = types.attrsOf types.path;
```

Comment:
Well it could be a string or a path. Im unsure, but string feels somehow better.